bst.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+
+class Node(object):
+    """
+    A Node class that will create instances of nodes, all of which can
+    represent the root of a binary search tree.
+    """
+    def __init__(self, value):
+        """
+        Make a new nodes object with a value given as the argument
+        """
+        self.value = value
+        self.left = None    # initially set pointers to None
+        self.right = None
+
+    def insert(self, value):
+        """
+        Insert a new node into a tree with value given as argument if value
+        is not present, else do nothing
+        """
+        if value == self.value:
+            return   # if value exists, do nothing
+        elif value < self.value:
+            if self.left is None:
+                self.left = Node(value)  # create value at empty pointer
+            else:
+                return self.left.insert(value)  # repeat for next node
+        else:
+            if self.right is None:
+                self.right = Node(value)
+            else:
+                return self.right.insert(value)
+
+    def contains(self, value):
+        """
+        Check a tree for a node with value given as argument, return True
+        if the value is present, False if not
+        """
+        if value == self.value:
+            return True
+        elif value < self.value:
+            if self.left is None:
+                return False
+            else:
+                return self.left.contains(value)
+        else:
+            if self.right is None:
+                return False
+            else:
+                return self.right.contains(value)
+
+    def size(self):
+        """
+        Return the number of nodes in a tree
+        """
+        if self.left is None and self.right is None:
+            return 1
+        if self.left is None:
+            return self.right.size() + 1   # Add 1 to the sum of branch sizes
+        if self.right is None:             # to account for self
+            return self.left.size() + 1
+        return self.left.size() + self.right.size() + 1
```

Comment:
You have to add one no matter what, why do it four times?  And why four separate return statements?  Too complex.